src/components/BorrowerProfile/BorrowerName.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+	<h1 class="text-h2">
+		{{ name }}
+	</h1>
+</template>
+
+<script>
+import gql from 'graphql-tag';
+
+export default {
+	inject: ['apollo', 'cookieStore'],
+	data() {
+		return {
+			name: '',
+		};
+	},
+	apollo: {
```

Comment:
I don't think this should be in this component. The name shouldn't need it's own direct query.

Reply:
Could you talk through more of your reasoning for that? My thinking is that we have each component responsible for its own data. That way the components can be loosely coupled, we can selectively control what is and isn't prefetched, and if a component needs to be changed to include extra data that change only needs to happen in the component itself and no changes are needed in the wrapping component. I could see it maybe being overkill for the borrower name,  as it's a single data point, but I think in general we should have each component fetch its own data (especially things like the fundraising progress component, which will need several data points).

Reply:
@mcstover How about this:

- in a separate PR we make a "SummaryCard" component for the borrower profile that handles the data fetching and display of the avatar image, loan use, name, fundraising progress, and location/sector
- if we make child components for the SummaryCard, they just use props
- for this PR we change this component to accept name as a prop

I think that will allow us to not need this apollo boilerplate code for just the name, while also maintaining a loose coupling between most of the components on the page.

My main goal is to avoid a situation where we are passing around a single large loan object to every component on the page that only need one piece of data from the loan.

Reply:
I think the strategy you described above works for some collections of components/data but we should strike a better balance between significantly increasing the number of queries needed to bootstrap the page vs the independent data fetch.

Generally, the BP card section should prob have 1 query that gets the name, image, location, description + tag then a second query for the loan fundraising info + meter section.

This is effectively 7 vs 2 queries and we should consider the additional overhead that comes with the prefetch subscription which will run and update the data 2x for each query.

As for moving toward the more independently modular approach, I'd like to see some ways to remove the level of boilerplate required to bootstrap this. In this case that could mean passing the query and param values which currently depend on a specific id in the url which reduces the reusability.

Reply:
That sounds good @emuvente !!

Reply:
@mcstover Makes sense, and sounds good!
@mlibazisi sorry for the churn on this, could you please change this component back to accepting the name as a prop?

Reply:
👍